src/core/components/inputs/TextInput.tsx

```diff
@@ -0,0 +1,102 @@
+import React, {useEffect, useState} from 'react';
+import {StyledEyeBtn, StyledInputRowWrapper, StyledInputWrapper, StyledTextInput} from './Inputs.styled';
+import {
+    Keyboard,
+    StyleProp,
+    TextStyle,
+    ViewStyle
+} from "react-native";
+import Ionicons from "@expo/vector-icons/Ionicons";
+import {TextInput as RNTextInput} from "react-native";
+import {RegularText} from "@components/textComponents/TextComponents";
+
+type TextInputProps = React.ComponentProps<typeof RNTextInput> & {
+    value: string,
+    required?: boolean,
+    errorHandler?: (arg0: boolean) => void,
+    label?: string,
+    onChange: (value: string) => void,
+    isPassword?: boolean,
+    validation?: { rule: RegExp, message: string }[],
+    styles?: {
+        wrapper?: StyleProp<ViewStyle>,
+        input?: StyleProp<TextStyle>,
+        label?: StyleProp<TextStyle>,
+        error?: StyleProp<TextStyle>
+    }
+}
+
+const TextInput: React.FC<TextInputProps> = (props) => {
+    const {
+        required = false,
+        value,
+        errorHandler,
+        isPassword,
+        validation,
+        onChange = (value: string) => null,
+        styles,
+        label,
+        ...rest
+    } = props;
+
+    const [internalValue, setInternalValue] = useState(value)
+    const [error, setError] = useState('')
+    const [isFocused, setFocus] = useState(false)
+    const [secureText, setSecureText] = useState(isPassword);
+
+    useEffect(() => {
+        if (required && !value) {
+            errorHandler && errorHandler(true)
+        }
```

Comment:
Will it show errors? Bad UX. Did u see somewhere behaviour like that? User just opens the screen and the app already tells him/her that user did something wrong.